### DISCUSSION TITLE
Request Verification

### DISCUSSION THREAD

**Discussion by ElliotHYLee** [pocketbase/pocketbase#2888] [Q&A] [answered]
1. It looks it is public..?  /api/collections/users/request-verification
It may need to require a user or admin token. If I know the server is PB and any one of the user emails, I can send daily email to attack the email owner by spamming.

2. Does the PB server's API accepter has  rate limiter for this api?
I'm using mailgun. From the PB request logs, it receives the request from Potman. But the email doesn't arrive everytime. However, if I send the test email several times continuously, no issue.

**Comment by ganigeorgiev** [comment, marked as answer]
1. The verification url is public by default. The idea is that you can request a verification before authenticating. The verification email is sent **only for unverified users**. Additionally, we have a basic flood protection that prevents sending the same verification email if the previous email was send <= 2 mins ago. The same apply for the password reset request.

2. Yes, only for the public endpoints that send emails. General API rate limiting may get implemented in the future as it was requested by several users (see https://github.com/pocketbase/pocketbase/issues/1232) but for now remains a low priority.

**Comment by iamveso** [comment]
Glad I came across this discussion. Been trying to figure out why my tests werent working. turns out the email was already verified. I'd have to change my forgot-password flow but this was really helpful.
